### PR TITLE
Set vm name at creation time (directly in template) rather than renaming after creation.

### PR DESCRIPTION
**Pull request by benosman**
I noticed that this provider currently creates vms without a name and then renames the vm after creation. I happen to use the variable $NAME in my templates, so that behaviour causes issues as it retrieves the computed name instead (e.g. one-vm-1, one-vm-2, etc.)

Is there a specific reason why this approach is taken, rather than adding the name value into the template?

I've tested out that way and it works for me so far. I've added a pull request with those changes if this is something that can be considered.

### CODE REVIEW

**Comment by benosman** [comment]
Will probably conflict with  #13 if that gets added.

**Comment by treywelsh** [comment]
Hi @benosman ,

Thank you for spending time on the provider.

I don't know why it's done it this way (I'm not the initial author of the provider) but I understand your problem.

As I don't want to break something that I didn't aware of, I prefer to wait a bit for a response from @jaypif 
We just have to wait some days as he's off for now, sorry for that.

If it's of interest, don't worry, I'll make the same changes on top of the PR #13

File: opennebula/resource_opennebula_virtual_machine.go [hunk 1]
```diff
@@ -832,6 +825,9 @@ func waitForVmState(d *schema.ResourceData, meta interface{}, state string) (int
 
 func generateVmXML(d *schema.ResourceData) (string, error) {
 
+	// Retrieve Name
+	name := d.Get("name").(string)
```
Comment (jaypif) [inline opennebula/resource_opennebula_virtual_machine.go:829]: name is not mandatory on OpenNebula. If it is not set is use the one generated by OpenNebula.
So in your case you will have VM with empty name if you don't provide it in the template neither than in the `tf` file.

**Review by jaypif — CHANGES_REQUESTED** [review]
Hi @benosman 

Sorry for the late reply

Actually "name" is not mandatory because it is not the case on OpenNebula. If you create a VM without a name, OpenNebula gives you a generic one.
But you are right, you issue is totally legit.

However your fix should support the case the name is not set in the `tf` neither than in the template.

And can you also rebase your future fix on master (to keep tree quite clear)

**Comment by treywelsh** [comment]
Changes reported in #13 
Feel free to open an issue if something wrong :)

**Pull request closed without merging**